src/pydata_sphinx_theme/assets/styles/variables/_color.scss

```diff
@@ -1,21 +1,46 @@
+/* for colored buttons and badges, ensure text color is legible */
+@function text-color($value) {
+  @return if(lightness($value) > 70, #000, #fff);
+}
+
 /*******************************************************************************
 * light theme
 *
 * all the variables used for light theme coloring
 */
 html[data-theme="light"] {
   /*****************************************************************************
-  * main colors
+  * main colors. the $pst-semantic-colors follow the naming convention in
+  * sphinx-design, to allow for color overrides in
+  * ../extensions/_sphinx_design.scss. NOTE: some of these colors (most notably
```

Comment:
See my other point about intermixing sphinx-design and theme-specific colors, but if we're going to do it this way, can we be remove the uncertainty here? E.g., not "some of these colors", but instead list the specific colors this statement applies to. And remove the "probably", we should be precise and explicit because others without context will have a hard time reasoning what is a sphinx-design thing and what's a theme thing, as well as the implications of using those variables in the theme in general.

Sorry for being a stickler on this, but we recently spent a lot of effort trying to reduce our surface area of variables that are exposed to devs and to users, so just trying to make sure we stick with those goals :-)

Reply:
> can we remove the uncertainty here?

This point is now moot; all colors needed for sphinx-design overrides are now *only* in `extensions/_sphinx_design.scss`